### DISCUSSION TITLE
Electrical impacts factors

### DISCUSSION THREAD

**Discussion by pmdirector** [Boavizta/boaviztapi#223] [Q&A] [answered]
This ref (https://dev.doc.api.boavizta.org/Explanations/usage/elec_factors/) says users can provide their own impact factors.
This ref (https://dev.api.boavizta.org/docs#/cloud/instance_cloud_impact_v1_cloud_instance_post) says use the following format.
Impact factors can be given : "usage":{"elec_factors":{[criterion]_factors: 0.38}}

So if I wanted to change 0.38 kgCO2e/kWh to 0.41 kgCO2e/kwh (say), would I use a schema like the following?
{
  "provider": "aws",
  "instance_type": "a1.4xlarge",
  "usage": {
    "elec_factors": {gwp_factors: 0.41},
    "time_workload": [
      {
        "time_percentage": 50,
        "load_percentage": 0
      },
      {
        "time_percentage": 25,
        "load_percentage": 60
      },
      {
        "time_percentage": 25,
        "load_percentage": 100
      }
    ]
  }
}

One last thing: the first ref above has a warning not to use market-based factors. Why is that? Some CSPs are purchasing all their energy from renewable sources in regions where the average carbon intensity is not zero.

Thanks for any guidance.

**Comment by da-ekchajzer** [comment, marked as answer]
First of all, thank you for your question. 

## Format

Concerning the format, there is an error in the swagger documentation. The correct format can be found here: https://dev.doc.api.boavizta.org/Reference/format/usage/#general

```
{
 "usage": {
   "usage_location": "FRA",
   "avg_power": 120,
   "elec_factors": {
     "gwp": 0.1
   }
 }
}

```

## Market-based approach

> One last thing: the first ref above has a warning not to use market-based factors. Why is that? Some CSPs are purchasing all their energy from renewable sources in regions where the average carbon intensity is not zero.

There are many reasons not to consider those figures when you use the API. 

### Normative argument

* GHG protocol recommend that company disclose both local-based and market-based see : https://www.dnv.co.uk/services/scope-2-dual-reporting-65855

* The Bilan Carbone® doesn't consider the market-based approach : 

> Bilan Carbone® is intended to reflect a physical reality, not a commercial appellation. Renewable electricity production is rarely self-consumed, and is therefore very often connected to the national grid: this "green" production is therefore part of the national electricity mix. The development of "green" energies means a reduction in the emission factor of the national electricity mix.

### Other arguments

* Most renewable energy investment is not self-consumed. Market-based doesn't reflect a physical reality but a market reality. For instance, when the electricity demand is high, datacenters will consume energy from fossil fuel.

* If you apply a smaller emission factor to cloud providers, it means that should apply a bigger emission factor to other activities to make sure you don't forget any emissions. This typically leads to greater impacts on people's heating systems, or on less profitable industries that can't afford to buy renewable energy certificates. This raises a question of fairness

The market-based metric may be interesting for describing investments, but it should not be used in an environmental accounting context.

**Reply by pmdirector** [reply]
Thanks for the response. Allowing the user to input their own electrical impact factors is essential in my opinion.

**Reply by da-ekchajzer** [reply]
This is indeed the case if you have impact factors from more up-to-date databases or on a specific territory. However, I maintain that market-based impact factors, although they may be part of providers' communications, are not designed to be reused in customers' environmental accounting.
